haddock-api/src/Haddock/Interface.hs

```diff
@@ -120,34 +122,55 @@ processModules verbosity modules flags extIfaces = do
 -- * Module typechecking and Interface creation
 --------------------------------------------------------------------------------
 
-
 createIfaces :: Verbosity -> [String] -> [Flag] -> InstIfaceMap -> Ghc ([Interface], ModuleSet)
 createIfaces verbosity modules flags instIfaceMap = do
   -- Ask GHC to tell us what the module graph is
   targets <- mapM (\filePath -> guessTarget filePath Nothing) modules
   setTargets targets
-  modGraph <- depanal [] False
+  _modGraph <- depanal [] False
 
-  -- Visit modules in that order
-  let sortedMods = flattenSCCs $ topSortModuleGraph False modGraph Nothing
   out verbosity normal "Haddock coverage:"
-  (ifaces, _, !ms) <- foldM f ([], Map.empty, emptyModuleSet) sortedMods
-  return (reverse ifaces, ms)
+
+  ifaceMapRef <- liftIO $ newIORef (Map.empty, emptyModuleSet)
+  loadOk <- {-# SCC load #-}
+            withTiming getDynFlags "load" (const ()) $
+              gbracket_
+                (modifySession $ setHooks $ emptyHooks
+                  { hscFrontendHook = Just $ hook ifaceMapRef
+                  })
+                (modifySession $ setHooks emptyHooks) $
+                GHC.load LoadAllTargets
+  case loadOk of
+    GHC.Succeeded -> do
+      (ifaceMap, !ms) <- liftIO $ atomicModifyIORef' ifaceMapRef $ \i ->
+        ((Map.empty, emptyModuleSet), i)
+      return (Map.elems ifaceMap, ms)
```

Comment:
There was some filtering here, but they seems not required

Reply:
What "filtering" do you refer to?

Reply:
https://github.com/facebook/fbhaddock/pull/47/commits/e01cccf0773da48ce930d7765a2d733bffed4233#diff-a6119dd2da927bbff2c39eda86cdc733R149-R152